sr/comp/cli/show_match_scores.py

```diff
@@ -0,0 +1,233 @@
+                tla=None,
+                ranking=None,
+                game=None,
+                league=None,
+            )
+
+    return MatchResult(
+        num=match.num,
+        arena=match.arena,
+        display_name=match.display_name,
+        corners=corner_data,
+    )
+
+
+def match_index(matches: List[MatchSlot], match_num: int) -> int:
+    "Returns the index of the first slot that contains the given match number"
+    for idx, slots in enumerate(matches):
+        for match in slots.values():
+            if match.num == match_num:
+                return idx
+
+    # if no match is found use the last one
+    return idx
+
+
+def generate_displayed_headings(num_corners: int) -> List[str]:
+    displayed_heading = ["Match"]
+
+    for _ in range(num_corners):
+        displayed_heading.append("Zone")
+        displayed_heading.append("TLA")
+        displayed_heading.append("Rank")
+        displayed_heading.append("Game")
+        displayed_heading.append("League")
+
+    return displayed_heading
+
+
+def generate_displayed_match(match: MatchResult, num_corners: int) -> List[List[str]]:
+    displayed_match = []
+    displayed_corners = []
+
+    for zone, (tla, ranking, game, league) in match.corners.items():
+        displayed_corner: List[str] = []
+
+        displayed_corner.append(str(zone))
+        if tla is not None:
+            displayed_corner.append(tla)
+            displayed_corner.append("??" if ranking is None else str(ranking))
+            displayed_corner.append("??" if game is None else str(game))
+            displayed_corner.append("??" if league is None else str(league))
+        else:
+            displayed_corner.extend(['', '', '', ''])
+
+        displayed_corners.append(displayed_corner)
+
+    # wrap the number of zones to the DISPLAYED_ZONES constant
+    for corner in range(0, num_corners, DISPLAYED_ZONES):
+        # first row displays the match and arena information,
+        # any extra rows leave this field blank
+        if corner == 0:
+            match_row = [f"{match.display_name} in {match.arena}"]
+        else:
+            match_row = [""]
+
+        for idx in range(DISPLAYED_ZONES):
+            try:
+                match_row.extend(displayed_corners[corner + idx])
+            except IndexError:
+                # pad the number of corners out to a multiple of DISPLAYED_ZONES
+                match_row.extend(['', '', '', ''])
+
+        displayed_match.append(match_row)
+
+    return displayed_match
+
+
+def command(settings):
+    import os.path
+
+    from tabulate import tabulate
+
+    comp = SRComp(os.path.realpath(settings.compstate))
+
+    match_results: List[MatchResult] = []
+
+    filter_tla = settings.tla
+    skip_filter = True
+
+    if filter_tla is not None:
+        skip_filter = False
+        # validate TLA exists
+        if filter_tla not in comp.teams.keys():
+            print('TLA not found')
+            return
```

Comment:
Since this is an error case, please could we:
- include the "bad" data in the message,
- put the error message on `stderr`, and
- return an error code so that (scripted) callers know something went wrong

I think we have some handlers for this sort of thing in the deploy script if you want some inspiration.
Aside: I've long been meaning to formalise this sort of thing though, likely styled after Django's `CommandError` exception, though that's likely overkill to introduce in this PR.

One approach here might be:
```suggestion
            exit(f"TLA {filter_tla!r} not recognised")
```